homeassistant/components/google/__init__.py

```diff
@@ -198,6 +235,65 @@ def _scan_for_calendars(service):
 
     hass.services.register(
         DOMAIN, SERVICE_SCAN_CALENDARS, _scan_for_calendars)
+
+    def _add_event(call):
+        """Add a new event to calendar."""
+        service = calendar_service.get()
+        start = {}
+        end = {}
+
+        if EVENT_IN in call.data:
+            if EVENT_IN_DAYS in call.data[EVENT_IN]:
+                now = datetime.now()
+
+                start_in = now + timedelta(
+                    days=call.data[EVENT_IN][EVENT_IN_DAYS])
+                end_in = start_in + timedelta(days=1)
+
+                start = {'date': start_in.strftime('%Y-%m-%d')}
+                end = {'date': end_in.strftime('%Y-%m-%d')}
+
+            elif EVENT_IN_WEEKS in call.data[EVENT_IN]:
+                now = datetime.now()
+
+                start_in = now + timedelta(
+                    weeks=call.data[EVENT_IN][EVENT_IN_WEEKS])
+                end_in = start_in + timedelta(days=1)
+
+                start = {'date': start_in.strftime('%Y-%m-%d')}
+                end = {'date': end_in.strftime('%Y-%m-%d')}
+
+        elif EVENT_START_DATE in call.data:
+            start = {'date': str(call.data[EVENT_START_DATE])}
+            end = {'date': str(call.data[EVENT_END_DATE])}
+
+        elif EVENT_START_DATETIME in call.data:
+            start_dt = str(call.data[EVENT_START_DATETIME]
+                           .strftime('%Y-%m-%dT%H:%M:%S'))
+            end_dt = str(call.data[EVENT_END_DATETIME]
+                         .strftime('%Y-%m-%dT%H:%M:%S'))
+            start = {'dateTime': start_dt,
+                     'timeZone': str(hass.config.time_zone)}
+            end = {'dateTime': end_dt,
+                   'timeZone': str(hass.config.time_zone)}
+
+        event = {
+            'summary': call.data[EVENT_SUMMARY],
+            'description': call.data[EVENT_DESCRIPTION],
+            'start': start,
+            'end': end,
+        }
+        service_data = {'calendarId': call.data[EVENT_CALENDAR_ID],
+                        'body': event}
+        event = (
+            service.events()
```

Comment:
they should fit on one line now? E.G. service.events().insert(**service_data).execute(). Or maybe I'm misunderstanding this part?

Reply:
No, it's me. Got to excited about the **kwargs bit that I forgot to move it up.